src/types/missing_scene.ts

```diff
@@ -13,23 +8,63 @@ export default class MissingScene {
 }
 
 import { sceneCollection, missingSceneCollection } from "../database/index";
+import Marker from "../types/marker";
+import { removeSceneFromQueue } from "../queue/processing";
+import LabelledItem from "../types/labelled_item";
+import ActorReference from "../types/actor_reference";
+import MovieScene from "../types/movie_scene";
+import Image from "../types/image";
+import { index as sceneIndex } from "../search/scene";
 import * as logger from "../logger";
 export async function purgeMissingScenes() {
   const items = await missingSceneCollection.getAll();
-  items.forEach(async item => {
+  for (const [key, item] of items.entries()) {
+    logger.log(`Deleting missing scene ${item.path}`);
+
     await sceneCollection
       .remove(item._id)
       .catch(err =>
         logger.error(
-          `Failed to remove ${item._id} at path ${item.path} from the db. Error: ${err}`
+          `Failed to remove ${item._id} at path ${item.path} from the sceneCollection. Error: ${err}`
         )
       );
+    await sceneIndex.remove([item._id]);
+    await Image.filterScene(item._id);
+
+    // deletes associated images when user wishes.
+    // No user option exists yet, enable this once user options are passed.
+    if (false) {
+      for (const image of await Image.getByScene(item._id)) {
+        await Image.remove(image);
+        await LabelledItem.removeByItem(image._id);
+      }
+      logger.success("Deleted images of scene " + item._id);
+    }
+    await Marker.removeByScene(item._id);
+    await LabelledItem.removeByItem(item._id);
+    await ActorReference.removeByItem(item._id);
+    await MovieScene.removeByScene(item._id);
+
+    logger.log("Deleting scene from queue (if needed)");
+    try {
+      await removeSceneFromQueue(item._id);
+    } catch (err) {}
```

Comment:
I'm assuming an error removing from the queue is not important since this is also what the copied code does. For clarity down the road, just add a comment explaining it
```suggestion
    } catch (err) {
      // Do nothing, does not matter if this fails
    }
```

Reply:
this is copy pasted from boii's code